companion/lib/Surface/USB/MiraboxStreamDock.ts

```diff
@@ -0,0 +1,1414 @@
+/*
+ * This file is part of the Companion project
+ * Copyright (c) 2025 Dorian Meid
+ * Authors: Dorian Meid <dnmeid@gmx.net>
+ *
+ * This program is free software.
+ * You should have received a copy of the MIT licence as well as the Bitfocus
+ * Individual Contributor License Agreement for companion along with
+ * this program.
+ *
+ * You can be released from the requirements of the license by purchasing
+ * a commercial license. Buying such a license is mandatory as soon as you
+ * develop commercial activities involving the Companion software without
+ * disclosing the source code of your own applications.
+ *
+ */
+
+import { EventEmitter } from 'events'
+import { Device, HIDAsync } from 'node-hid'
+import util from 'util'
+import imageRs from '@julusian/image-rs'
+import jpg from '@julusian/jpeg-turbo'
+import LogController, { Logger } from '../../Log/Controller.js'
+import { ImageWriteQueue } from '../../Resources/ImageWriteQueue.js'
+import { transformButtonImage } from '../../Resources/Util.js'
+import {
+	OffsetConfigFields,
+	BrightnessConfigField,
+	LegacyRotationConfigField,
+	LockConfigFields,
+} from '../CommonConfigFields.js'
+import type { CompanionSurfaceConfigField, GridSize } from '@companion-app/shared/Model/Surfaces.js'
+import type { SurfacePanel, SurfacePanelEvents, SurfacePanelInfo } from '../Types.js'
+import type { ImageResult } from '../../Graphics/ImageResult.js'
+
+const setTimeoutPromise = util.promisify(setTimeout)
+
+export class SurfaceUSBMiraboxStreamDock extends EventEmitter<SurfacePanelEvents> implements SurfacePanel {
+	public logger: Logger
+
+	config: Record<string, any> = {}
+
+	readonly #streamDock: StreamDock
+
+	readonly #writeQueue: ImageWriteQueue<string, [number, number, ImageResult]>
+
+	/**
+	 * Whether to cleanup the deck on quit
+	 */
+	#shouldCleanupOnQuit = true
+
+	readonly info: SurfacePanelInfo
+	readonly gridSize: GridSize
+
+
+	constructor(devicePath: string, streamDock: StreamDock) {
+		super()
+
+		this.logger = LogController.createLogger(`Surface/USB/MiraboxStreamDock/${devicePath}`)
+
+		this.config = {
+			brightness: 100,
+			rotation: 0,
+		}
+
+		this.#streamDock = streamDock
+
+		this.logger.debug(`Adding Mirabox ${this.#streamDock.productName} device: ${devicePath}`)
+
+		const configFields: CompanionSurfaceConfigField[] = [...OffsetConfigFields, BrightnessConfigField, LegacyRotationConfigField, ...LockConfigFields]
```

Comment:
I think this should use `RotationConfigField` instead of `LegacyRotationConfigField`.
The legacy version exists to avoid breaking existing configs, and should probably be phased out at some point. I don't think it makes sense to use for new surface types